paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -169,13 +173,70 @@ public class PaparazziPlugin @Inject constructor(
       val isRecordRun = project.objects.property(Boolean::class.java)
       val isVerifyRun = project.objects.property(Boolean::class.java)
 
+      val xmlTestOutputDir = project.objects.directoryProperty()
+      val htmlTestReportOutputDir = project.objects.directoryProperty()
+
       project.gradle.taskGraph.whenReady { graph ->
         isRecordRun.set(recordTaskProvider.map { graph.hasTask(it) })
         isVerifyRun.set(verifyTaskProvider.map { graph.hasTask(it) })
+
+        // Unfortunately, we have to wait till the task graph is ready. When using a custom reporting base directory
+        // the actual report directory gets overwritten after task configuration. Seems like AGP is not respecting this extensions config.
+        // So we wait to set the testReportOutputDir until the task graph is ready.
+        htmlTestReportOutputDir.set(testTaskProvider.firstOrNull()?.reports?.html?.outputLocation?.get())
```

Comment:
please don't add more to this callback, instead let's use a lambda/property to defer resolution.

Reply:
Also, can you elaborate more on "Seems like AGP is not respecting this extensions config".  In other words, what is AGP actually doing here?  I would guess they're not using `taskGraph.whenReady`.